### PR TITLE
Fixing some AVC denies for /var/log/vault_audit.log and 

### PR DESCRIPTION
**Pull request by eric-eisenhart**
On CentOS7, we saw a ton of this in /var/log/audit/audit.log when vault started up:

```
type=AVC msg=audit(1643241528.567:61295): avc:  denied  { open } for  pid=7939 comm="vault" path="/var/log/vault_audit.log" dev="dm-0" ino=134452775 scontext=system_u:system_r:vault_t:s0 tcontext=unconfined_u:object_r:var_log_t:s0 tclass=file permissive=0
```

As well as a variety of other AVC denials around dgram sockets for logging... And some of the outgoing TCP connection stuff also didn't work...

### CODE REVIEW

**Review by xntrik — COMMENTED** [review]
Seems all good to me! Will reach out to the HC sec team for some 👀

File: products/vault_selinux/vault.fc [hunk 1]
```diff
@@ -2,3 +2,4 @@
 /opt/vault(/.*)?		gen_context(system_u:object_r:vault_sys_content_t,s0)
 /etc/vault\.d(/.*)?		gen_context(system_u:object_r:vault_conf_t,s0)
 /var/log/vault(/.*)?		gen_context(system_u:object_r:vault_log_t,s0)
+/var/log/vault_audit_log	gen_context(system_u:object_r:vault_log_t,s0)
```
Comment (JAORMX) [inline products/vault_selinux/vault.fc:5]: wouldn't this be `/var/log/vault_audit\.log`? Do you have a rotation policy for your those audit logs? Depending on what's being used, something like:
```
/var/log/vault_audit\.log(\..*)?	gen_context(system_u:object_r:vault_log_t,s0)
```
Might be needed to also target any rotated files.

Comment (eric-eisenhart) [reply products/vault_selinux/vault.fc:5]: Oh, yeah, `.log`, not `_log`...  Not sure where I saw `_log`.

Maybe should just be `/var/log/vault(.*)`, or drop this and change `/var/log/vault(/.*)?` to `/var/log/vault(.*)?`

Comment (JAORMX) [reply products/vault_selinux/vault.fc]: I think using `/var/log/vault(.*)?` would be fine.

**Review by JAORMX — APPROVED** [review]

File: products/vault_selinux/vault.te [hunk 2]
```diff
@@ -59,8 +61,11 @@ allow vault_t self:fifo_file rw_fifo_file_perms;
 allow vault_t self:unix_stream_socket create_stream_socket_perms;
 # and network sockets too
 allow vault_t self:netlink_route_socket { bind create getattr nlmsg_read };
-allow vault_t self:unix_dgram_socket { bind create getattr setopt };
+allow vault_t self:unix_dgram_socket { bind create getattr setopt connect sendto };
 allow vault_t self:tcp_socket { accept bind create getattr listen setopt read write };
+allow vault_t unreserved_port_t:tcp_socket name_connect;
```
Comment (cipherboy) [inline products/vault_selinux/vault.te:66]: @eric-eisenhart, do you know what denials caused these additions? My understanding is you could trigger them with something like an external KMS or potentially a database credentials secrets backend? [`unreserved_port_t`](https://wiki.gentoo.org/wiki/SELinux/Tutorials/Managing_network_port_labels) is a broad range of ports, so my one hesitation is perhaps there's a small-ish set of more descriptive labels here that we could document better why we want these specific ports (databases, &c). But then again perhaps not worth it and perhaps better to add a comment describing why this was suggested :-)

**Review by cipherboy — APPROVED** [review]
Thanks for the review @JAORMX and @xntrik  :-)

**Comment by cipherboy** [comment]
:-) Finally have merge permission, sorry about delay.

**Pull request merged**